scripts/src/java/org/oppia/android/scripts/common/CommandExecutorImpl.kt

```diff
@@ -33,17 +43,57 @@ class CommandExecutorImpl(
         .directory(workingDir)
         .redirectErrorStream(includeErrorOutput)
         .start()
-    val finished = runBlocking {
-      CoroutineScope(scriptBgDispatcher).async {
-        process.waitFor(processTimeout, processTimeoutUnit)
-      }.await()
+
+    // Consume the input & error streams individually, and separately from waiting for the process
+    // to complete (since consuming the output channels may be required for the process to actually
+    // finish executing properly).
+    val stdoutLinesDeferred = process.inputStream.readAllLinesAsync()
+    val stderrLinesDeferred = process.errorStream.readAllLinesAsync()
```

Comment:
TODO: explain these executor changes (& why they're why the script background executor was added in the first place).

Reply:
Done.